Modules/SubModule_Import/xASL_imp_DCM2NII_Subject_ShuffleTheDynamics.m

```diff
@@ -130,40 +132,59 @@
-                if isHadamardFME
-                    % Check image
-                    if exist(nii_files{1} ,'file')
-                    	% Determine the number of time points within each NIfTI
-                        imASL = xASL_io_Nifti2Im(nii_files{1});
-						numberTEs = length(resultJSON.EchoTime);
-                        numberPLDs = int32(size(imASL,4)/numberTEs);
-						
-						% Reorder TEs and PLDs - first cycle TE afterwards PLD
-						vectorOldOrder = zeros(size(imASL,4),1);
-						for iPLD = 1:(double(numberPLDs))
-							vectorOldOrder((1:numberTEs)+(iPLD-1)*numberTEs) = (iPLD-1)+1:numberPLDs:size(imASL,4);
-						end
-						imASL(:,:,:,1:end) = imASL(:,:,:,vectorOldOrder);
-						xASL_io_SaveNifti(nii_files{1},nii_files{1},imASL);
-						% Repeat Echo Times
-						resultJSON.EchoTime = repmat(resultJSON.EchoTime,numberPLDs,1);
-						% Save the JSON with the updated echo times
-						spm_jsonwrite(fullfile(resultPath, [resultFile '.json']),resultJSON);
-                    else
-                    	% Fallback (something went wrong)
-						warning('FME Hadamard sequence with 1 PLD only');
+            if isfield(resultJSON,'SeriesDescription') || bHadamard
+            	% Determine if we have the specific FME Hadamard sequence from Bremen
+                bHadamardFME = ~isempty(regexp(char(resultJSON.SeriesDescription),'(Encoded_Images_Had)\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once'));
+                % If the FME sequence was detected we can always set the general bHadamard to true as well
+                if bHadamardFME
+                    bHadamard = true;
+                end
+            end
+            % Check if we the current sequence is a Hadamard or not
+            if bHadamard || bHadamardFME
+                % Check image
+                if xASL_exist(nii_files{1},'file')
+                    % Determine the number of time points within each NIfTI
+                    imASL = xASL_io_Nifti2Im(nii_files{1});
+                    numberTEs = length(resultJSON.EchoTime);
+                    numberPLDs = int32(size(imASL,4)/numberTEs);
+                    
+                    % Reorder TEs and PLDs - first cycle TE afterwards PLD
+                    vectorOldOrder = zeros(size(imASL,4),1);
+                    for iPLD = 1:(double(numberPLDs))
+                        vectorOldOrder((1:numberTEs)+(iPLD-1)*numberTEs) = (iPLD-1)+1:numberPLDs:size(imASL,4);
                     end
+                    imASL(:,:,:,1:end) = imASL(:,:,:,vectorOldOrder);
+                    xASL_io_SaveNifti(nii_files{1},nii_files{1},imASL);
+                    % Repeat Echo Times
+                    resultJSON.EchoTime = repmat(resultJSON.EchoTime,numberPLDs,1);
+                    % Save the JSON with the updated echo times
+                    spm_jsonwrite(fullfile(resultPath, [resultFile '.json']),resultJSON);
+                else
+                    % Fallback (something went wrong)
```

Comment:
The comment says "something went wrong" whereas the warning is specific "1 PLD only"). Is this correct?

Reply:
The comment is just older Henk...